src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs

```diff
@@ -580,8 +580,14 @@ internal void TogglePreviewControlAllowance()
 
             if (previewEnabled == false && !PreviewControl.StaysOpen)
             {
-                PreviewControl.TransitionToState(PreviewControl.State.Condensed);
-                PreviewControl.TransitionToState(PreviewControl.State.Hidden);
+                if (PreviewControl.IsExpanded)
+                {
+                    PreviewControl.TransitionToState(PreviewControl.State.Condensed);
```

Comment:
condensed and hidden? Hidden should always make the preview condense and hide. Think about it.